src/OctoshiftCLI.Tests/bbs2gh/Services/BbsArchiveDownloaderTests.cs

```diff
@@ -12,72 +13,82 @@ public sealed class BbsArchiveDownloaderTests : IDisposable
 {
     private const int EXPORT_JOB_ID = 1;
     private const string BBS_HOME_DIRECTORY = "BBS_HOME";
+    private const string TARGET_DIRECTORY = "TARGET";
 
     private readonly string _exportArchiveFilename = $"Bitbucket_export_{EXPORT_JOB_ID}.tar";
     private readonly Mock<OctoLogger> _mockOctoLogger = TestHelpers.CreateMock<OctoLogger>();
     private readonly Mock<FileSystemProvider> _mockFileSystemProvider = TestHelpers.CreateMock<FileSystemProvider>();
+    private readonly Mock<ISftpClient> _mockSftpClient = new();
     private readonly BbsArchiveDownloader _bbsArchiveDownloader;
-    private readonly ScpClient _scpClient;
 
     public BbsArchiveDownloaderTests()
     {
-        _scpClient = new ScpClient("host", "username", "password");
-        _bbsArchiveDownloader = new BbsArchiveDownloader(_mockOctoLogger.Object, _scpClient)
+        _bbsArchiveDownloader = new BbsArchiveDownloader(_mockOctoLogger.Object, _mockSftpClient.Object)
         {
             FileSystemProvider = _mockFileSystemProvider.Object,
             BbsSharedHomeDirectory = BBS_HOME_DIRECTORY
         };
+
+        _mockSftpClient.Setup(m => m.Exists(It.IsAny<string>())).Returns(true);
+
+        var mockAsyncResult = new Mock<IAsyncResult>();
+        mockAsyncResult.Setup(m => m.IsCompleted).Returns(true);
+        _mockSftpClient
+            .Setup(m => m.BeginDownloadFile(It.IsAny<string>(), It.IsAny<Stream>(), It.IsAny<AsyncCallback>(), It.IsAny<object>(), It.IsAny<Action<ulong>>()))
+            .Returns(mockAsyncResult.Object);
     }
 
     [Fact]
-    public void Download_Calls_ScpClient_Download_With_Correct_Params()
+    public async Task Download_Calls_SftpClinet_DownloadFile_With_Correct_Params()
     {
         // Arrange
         var expectedSourceArchiveFullName = Path.Combine(BBS_HOME_DIRECTORY, "data/migration/export", _exportArchiveFilename);
-
-        string actualSource = null;
-        string actualTarget = null;
-        _bbsArchiveDownloader.ScpDownload = (source, target) =>
-        {
-            actualSource = source;
-            actualTarget = target.Name;
-        };
+        var expectedTargetArchiveFullName = Path.Combine(TARGET_DIRECTORY, _exportArchiveFilename);
 
         // Act
-        _bbsArchiveDownloader.Download(EXPORT_JOB_ID);
+        await _bbsArchiveDownloader.Download(EXPORT_JOB_ID, TARGET_DIRECTORY);
 
         // Assert
-        actualSource.Should().Be(expectedSourceArchiveFullName);
-        actualTarget.Should().Be(_exportArchiveFilename);
+        _mockSftpClient.Verify(m =>
+            m.BeginDownloadFile(
+                It.Is<string>(actual => actual == expectedSourceArchiveFullName),
+                It.IsAny<Stream>(),
+                null,
+                null,
+                It.IsAny<Action<ulong>>()));
+
+        _mockFileSystemProvider.Verify(m => m.Open(It.Is<string>(actual => actual == expectedTargetArchiveFullName), FileMode.CreateNew));
```

Comment:
```suggestion
        _mockFileSystemProvider.Verify(m => m.Open(expectedTargetArchiveFullName, FileMode.CreateNew));
```

Reply:
addressed in 508e086bbd64c5766ba77cdd00b690d1ed1ec286